ats/pkb/pkb.py

```diff
@@ -55,27 +59,70 @@ def on_node_enter(node: nodes.ASTNode, context: dict):
                 stmt_index[-1] += 1
                 stmt_id += 1
 
+                stack.append(node.__stmt_id)
+
+                # determining the order of call
+                if isinstance(node, nodes.StmtCallNode):
+                    if node.name in proc_parents:
+                        proc_parents[node.name].extend(stack)
+                    else:
+                        proc_parents[node.name] = stack[:]
+
+                    if stack[0] not in call_order and node.name not in call_order:
+                        call_order.append(stack[0])
+                        call_order.append(node.name)
+                    elif stack[0] not in call_order and node.name in call_order:
+                        call_order.insert(call_order.index(node.name), stack[0])
+                    elif stack[0] in call_order and node.name not in call_order:
+                        call_order.insert(call_order.index(stack[0]) + 1, node.name)
+                    else:
+                        if call_order.index(node.name) < call_order.index(stack[0]):
+                            call_order.insert(call_order.index(stack[0]) + 1, node.name)
+                        if call_order.index(node.name) > call_order.index(stack[0]):
+                            call_order.insert(call_order.index(node.name), stack[0])
```

Comment:
co to robi? Czemu jest tyle rozgalezien ifa?

Reply:
jest tutaj tyle ifów żeby ustalić w jakiej kolejności wywoływane są funkcje np
```
proc 1 {
  call 2
}
proc 2 {
  call 4
}
proc 3 {
  call 1
}
proc 4 {
  ...
}
```
i kolejnośc to 4, 2, 1, 3

Reply:
To jest wazne w ogole?

Reply:
Tak bo musimy wiedzieć w jakiej kolejności rozszerzać rodziców i w ogóle to źle wyżej napisałem kolejność to 3, 1, 2, 4
i rozszerzamy 1 o 3 potem 2 o 1 która już jest rozszerzona o 3 i 4 o 2 czyli też 3 i 1.

Reply:
czy ogolnie, do uses lub modifies jest potrzebne to czy nie? https://xyproblem.info

Reply:
jest potrzebne i do uses i modifies